### PR TITLE
feat(action): autogenerate the operator yaml with the image tag

### PR DESCRIPTION
**Pull request by pawanpraka1**
Signed-off-by: Pawan <pawan@mayadata.io>

fixes: https://github.com/openebs/zfs-localpv/issues/328

**Why is this PR required? What issue does it fix?**:

automatically update the operator yaml with the image tag at the release time.

### CODE REVIEW

**Review by shubham14bajpai — APPROVED** [review]
LGTM

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -56,10 +58,31 @@ jobs:
...
+          sed -i "s/zfs-driver:ci/zfs-driver:${RELEASE_TAG}/" deploy/yamls/zfs-driver.yaml
+          sed -i "s/zfs-driver:ci/zfs-driver:${RELEASE_TAG}/" deploy/zfs-operator.yaml
+          sed -i "s/newTag: ci/newTag: ${RELEASE_TAG}/" deploy/yamls/kustomization.yaml
+
+          git diff --quiet deploy/ || CHANGED=$?
+          if [[ $CHANGED -ne 0 ]]; then
+            echo "Commiting operator yaml changes"
+            git add deploy/
+            git commit -s -m 'chore(yaml): updating operator yaml with the image tag'
+            git push origin HEAD:"${BRANCH}"
```
Comment (akhilerm) [inline .github/workflows/release.yml:84]: @pawanpraka1 Was this tested?

The HEAD will be in a detached state, so how will this push work?

**Review by akhilerm — COMMENTED** [review]
Added a comment

File: .github/workflows/release.yml (hunk 1 above)
Comment (pawanpraka1) [reply .github/workflows/release.yml:84]: yes, I have tested it, it works. It will push to the target branch mentioned in the command.

**Review by akhilerm — APPROVED** [review]
LGTM

**Review by prateekpandey14 — APPROVED** [review]
lgtm

**Pull request merged**